app/controllers/api/v1/repos_controller.rb

```diff
@@ -32,20 +32,30 @@ def show
   # POST /api/v1/users/:user_id/repos
   def create
     user = User.find(params[:user_id])
-    repo = user.repos.new(repo_params)
 
-    if repo.save
-      render json: repo, status: :created
+    if user.repos.exists?(owner: repo_params[:owner], name: repo_params[:name])
+      render json: { error: "Repo already exists." }, status: :unauthorized
     else
```

Comment:
I haven't seen it done this way before. That is neat way to check for its existence first before doing the action.